server.ts

```diff
@@ -1,6 +1,8 @@
 import { Request, Response } from "express";
 
 import dotenv from "dotenv";
+dotenv.config(); // load .env variables
```

Comment:
was this intentionally left in?

Reply:
left in intentionally, for some reason this file gets executed before the server.ts vars are injected at runtime, probably due to some circular import thing which is a rabbit hole I did not want to go down. because of this, there were instances where the env var in this file was undefined. I'm sure there's a fix to this somewhere, will just take a crap load of time to debug

Reply:
would you prefer to not have this temporarily? doesn't really affect performance, it just looks a bit odd

Reply:
oh that's so weird, I wouldn't have thought it would make a difference.

This is 100% something I think a dev would change back in the future for style reasons alone - can you add a comment explaining the rationale for moving it above some of the imports?

Reply:
I have some time tomorrow, I'll try and look into getting rid of it, maybe we don't need it to be something in the backlog lol

Reply:
OKAY so did some research, this happens because all imports are hoisted to the top before any code is run (cuz of our es6 imports) so the `dotenv.config()` happens after we import the `./api` file that uses the variable

Reply:
I can fix this

Reply:
fixed